x/ibc/03-connection/types/version.go

```diff
@@ -139,37 +123,42 @@ func GetFeatureSetIntersection(sourceFeatureSet, counterpartyFeatureSet []string
 	return featureSet
 }
 
-// VerifyProposedFeatureSet verifies that the entire feature set in the
-// proposed version is supported by this chain.
-func VerifyProposedFeatureSet(proposedVersion, supportedVersion string) bool {
-	_, proposedFeatureSet, err := UnpackVersion(proposedVersion)
-	if err != nil {
-		return false
+// VerifyProposedVersion verifies that the entire feature set in the
+// proposed version is supported by this chain. If the feature set is
+// empty it verifies that this is allowed for the specified version
+// identifier.
+func VerifyProposedVersion(proposedVersion, supportedVersion Version) error {
```

Comment:
should this be a method of Version instead?

Reply:
this make sense to me, will give it a shot and see if there is any good reason why it shouldn't be